conans/test/functional/configuration/profile_test.py

```diff
@@ -635,3 +635,60 @@ def test_profile_crazy_inheritance(self):
                              compiler.runtime=MD
                              compiler.version=15
                              os=Windows"""), self.client.out)
+
+
+def test_profile_from_cache_path():
+    """ When passing relative folder/profile as profile file, it MUST be used
+        conan install . -pr=profiles/default
+        /tmp/profiles/default MUST be consumed as target profile
+        https://github.com/conan-io/conan/pull/8685
+    """
+    client = TestClient()
+    path = os.path.join(client.cache.profiles_path, "android", "profile1")
+    save(path, "[settings]\nos=Android")
+    client.save({"conanfile.txt": ""})
+    client.run("install . -pr=android/profile1")
+    assert "os=Android" in client.out
+
+
+def test_profile_from_relative_pardir():
+    """ When passing relative ../path as profile file, it MUST be used
+        conan install . -pr=../profiles/default
+        /tmp/profiles/default MUST be consumed as target profile
+    """
+    folder = temp_folder(False)
+    client = TestClient()
+    client.save({os.path.join(folder, "profiles", "default"): "[settings]\nos=AIX",
+                 os.path.join(folder, "current", "conanfile.txt"): ""})
+    with client.chdir(os.path.join(folder, "current")):
+        client.run("install . -pr=../profiles/default")
```

Comment:
```suggestion
    client = TestClient()
    client.save({"profiles/default": "[settings]\nos=AIX",
                      "current/conanfile.txt": ""})
    with client.chdir("current"):
        client.run("install . -pr=../profiles/default")
```

You can simplify the tests by using the TestClient current folder which is already a temporary one, intended for these cases.

Reply:
True, I forgot client.current_folder != client.cache_folder. Thanks